### DISCUSSION TITLE
Migrate postgres database to pocketbase ??

### DISCUSSION THREAD

**Discussion by amirkhalil22** [pocketbase/pocketbase#2014] [Q&A]
hey guys,
I want a solution of how can i migrate my database from postgres to pocketbase !!!!!
A+

**Comment by ganigeorgiev** [comment]
There is no automated solution for this and without any information about your existing schema and data, I'm not sure how to help.

To get a basic idea how PocketBase manages its data and what fields are available, please check https://pocketbase.io/docs/collections/.

My general recommendation would be to first adapt your existing structure to collections manually using the Admin UI.
And after you are satisfied with your collections configuration, you can create a SQLite import script based on the collections fields OR export your existing data as csv/json and manually insert them with the [JS SDK](https://github.com/pocketbase/js-sdk) by iterating over each entry.

**Reply by amirkhalil22** [reply]
i'am working with simple databse in postgres just for test, i'am really trynig to convert it to csv/json but i don't know after that how can i upload that file.
after importing my file this is the error,

![image](https://user-images.githubusercontent.com/104743918/223872366-364a417f-52c2-4ea9-9899-d9d0f9b0a305.png)

**Reply by ganigeorgiev** [reply]
The "Import collections" UI is only for importing collection configurations (aka. json similar to the one you get from "Export collections"), it is not for data.

**Reply by amirkhalil22** [reply]
ohh i understand so how can i import my json these data?? or there is any way !!!!

**Reply by ganigeorgiev** [reply]
There is no built-in web api for importing records as json (_see https://github.com/pocketbase/pocketbase/issues/48#issuecomment-1211775627_).

As a workaround, you can apply the suggestions from my earlier comment:

> And after you are satisfied with your collections configuration, you can create a SQLite import script based on the collections fields OR export your existing data as csv/json and manually insert them with the [JS SDK](https://github.com/pocketbase/js-sdk) by iterating over each entry.

**Reply by amirkhalil22** [reply]
thanks, a lot! 
Just one more thing i don't understand this step,
"export your existing data as csv/json and manually insert them with the [JS SDK](https://github.com/pocketbase/js-sdk) by iterating over each entry."

**Reply by ganigeorgiev** [reply]
Here is a [JS SDK](https://github.com/pocketbase/js-sdk) example:

```js
import PocketBase from "pocketbase";

const pb = new PocketBase("http://127.0.0.1:8090");

const items = [
    {name: "test1"},
    {name: "test2"},
    {name: "test3"},
    // ...
];


// if you want to preserve the insertion order, await each request
for (let item of items) {
    await pb.collection("example").create(item);
}

// ---

// or if you don't care about the insertion order, you can insert them concurrently
const promises = [];
for (let item of items) {
    promises.push(pb.collection("example").create(item, {
        "$autoCancel": false,
    }));
}
await Promise.all(promises);
```
